# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wgpu/wgpu-core/Cargo.toml
# Commit: ef5dc3e04e5f
# Full Hash: ef5dc3e04e5f271eea0636ab3a495e95cc912f1d
# Author: Dzmitry Malyshau <dmalyshau@mozilla.com>
# Date: 2021-09-04 09:40:24
# Regressor Bug: 1726626
# File Overlap Count: 3
# Description:
#   Bug 1726626 - Move gfx/wgpu into a 3rd party dependency r=jgilbert,bholley
#   
#   This update makes wgpu a vendored dependency instead of having it in gfx/wgpu.
#   
#   ## Notes
# ==============================================================================

diff --git a/gfx/wgpu/wgpu-core/Cargo.toml b/third_party/rust/wgpu-core/Cargo.toml
rename from gfx/wgpu/wgpu-core/Cargo.toml
rename to third_party/rust/wgpu-core/Cargo.toml
--- a/gfx/wgpu/wgpu-core/Cargo.toml
+++ b/third_party/rust/wgpu-core/Cargo.toml
@@ -1,74 +1,64 @@
 [package]
 name = "wgpu-core"
-version = "0.9.2"
+version = "0.10.0"
 authors = ["wgpu developers"]
 edition = "2018"
 description = "WebGPU core logic on gfx-hal"
 homepage = "https://github.com/gfx-rs/wgpu"
 repository = "https://github.com/gfx-rs/wgpu"
 keywords = ["graphics"]
 license = "MIT OR Apache-2.0"
 
 [lib]
 
 [features]
 default = []
-# Enable SPIRV-Cross
-cross = ["gfx-backend-metal/cross"]
 # Enable API tracing
-trace = ["ron", "serde", "wgt/trace", "arrayvec/serde"]
+trace = ["ron", "serde", "wgt/trace", "arrayvec/serde", "naga/serialize"]
 # Enable API replaying
-replay = ["serde", "wgt/replay", "arrayvec/serde"]
+replay = ["serde", "wgt/replay", "arrayvec/serde", "naga/deserialize"]
 # Enable serializable compute/render passes, and bundle encoders.
 serial-pass = ["serde", "wgt/serde", "arrayvec/serde"]
 
 [dependencies]
-arrayvec = "0.5"
+arrayvec = "0.7"
 bitflags = "1.0"
 copyless = "0.1"
 fxhash = "0.2"
 log = "0.4"
 parking_lot = "0.11"
 profiling = { version = "1", default-features = false }
 raw-window-handle = { version = "0.3", optional = true }
 ron = { version = "0.6", optional = true }
 serde = { version = "1.0", features = ["serde_derive"], optional = true }
 smallvec = "1"
 thiserror = "1"
 
-gpu-alloc = "0.4"
-gpu-descriptor = "0.1"
-
-hal = { package = "gfx-hal", version = "0.9" }
-gfx-backend-empty = { version = "0.9" }
-
-[target.'cfg(all(not(target_arch = "wasm32"), all(unix, not(target_os = "ios"), not(target_os = "macos"))))'.dependencies]
-gfx-backend-vulkan = { version = "0.9", features = ["naga"] }
-#gfx-backend-gl = { version = "0.9 }
-
-[target.'cfg(all(not(target_arch = "wasm32"), any(target_os = "ios", target_os = "macos")))'.dependencies]
-gfx-backend-metal = { version = "0.9" }
-#TODO: could also depend on gfx-backend-vulkan for Vulkan Portability
-
-[target.'cfg(all(not(target_arch = "wasm32"), windows))'.dependencies]
-gfx-backend-dx12 = { version = "0.9" }
-gfx-backend-dx11 = { version = "0.9" }
-gfx-backend-vulkan = { version = "0.9", features = ["naga"] }
-
-[target.'cfg(target_arch = "wasm32")'.dependencies]
-#gfx-backend-gl = { version = "0.9" }
-
 [dependencies.naga]
-version = "0.5"
-features = ["spv-in", "spv-out", "wgsl-in"]
+git = "https://github.com/gfx-rs/naga"
+rev = "93db57c"
+#version = "0.6"
+features = ["wgsl-in"]
 
 [dependencies.wgt]
 path = "../wgpu-types"
 package = "wgpu-types"
-version = "0.9"
+version = "0.10"
+
+[dependencies.hal]
+path = "../wgpu-hal"
+package = "wgpu-hal"
+version = "0.10.1"
 
-[dev-dependencies]
-loom = "0.3"
+[target.'cfg(all(not(target_arch = "wasm32"), any(target_os = "ios", target_os = "macos")))'.dependencies]
+hal = { path = "../wgpu-hal", package = "wgpu-hal", version = "0.10", features = ["metal"] }
+#Note: could also enable "vulkan" for Vulkan Portability
+
+[target.'cfg(all(not(target_arch = "wasm32"), unix, not(target_os = "ios"), not(target_os = "macos")))'.dependencies]
+hal = { path = "../wgpu-hal", package = "wgpu-hal", version = "0.10", features = ["vulkan", "gles", "renderdoc"] }
+
+[target.'cfg(all(not(target_arch = "wasm32"), windows))'.dependencies]
+hal = { path = "../wgpu-hal", package = "wgpu-hal", version = "0.10", features = ["vulkan", "dx12", "renderdoc"] }
 
 [build-dependencies]
 cfg_aliases = "0.1"